Docs/570 install instructions (#578)
stan-dev/cmdstanpy: docsrc/installation.rst

Installation
============

CmdStanPy is a pure-Python3 package which wraps CmdStan,

> the command-line interface to Stan which is written in C++,
> therefore, in addition to Python3,

Reviewer: ```suggestion
the command-line interface to Stan which is written in C++.
Therefore, in addition to Python 3.7 or greater,
```

  GitHub `CmdStanPy <https://github.com/stan-dev/cmdstanpy>`_ repository.

If you install CmdStanPy from PyPI or GitHub you will need to
install CmdStan as well, see section :ref:`CmdStan Installation <cmdstan-install>` below.

  
Conda: install CmdStanPy, CmdStan, C++ toolchain
------------------------------------------------

If you use `conda <https://docs.conda.io/en/latest/>`_,
you can install CmdStanPy and the underlying CmdStan components from the
`conda-forge <https://conda-forge.org/>`_ repository
via the following command:

.. code-block:: bash

    conda create -n stan -c conda-forge cmdstanpy

This command creates a new conda environment named ``stan`` and 
downloads and installs the ``cmdstanpy`` package from conda-forge.

To install into an existing environment, use the conda ``install`` command instead of ``create``:

.. code-block:: bash

    conda install -c conda-forge cmdstanpy

Whichever installation method you use, afterwards you must
activate the new environment or deactivate/activate the existing one.
For example, if you installed cmdstanpy into a new environment ``stan``,
run the command

.. code-block:: bash

    conda activate stan

> The conda package manager will install CmdStan and the C++ toolchain.

Reviewer: I would move this line up to above the activate command so it is paired with the note on installing cmdstanpy

By default, the latest release of CmdStan is installed.
If you require a specific release of CmdStan, CmdStan versions
2.26.1 and *newer* can be installed by specifying
``cmdstan==VERSION`` in the install command.

> Versions before 2.26.1 can be downloaded from the CmdStan

Reviewer: ```suggestion
Versions before 2.26.1 are not available from conda but can be downloaded from the CmdStan
```

CmdStan Installation
--------------------

If you have installed CmdStanPy from PyPI or Github,
**you must install CmdStan**.
The recommended way to do so is via the ``install_cmdstan`` function
:ref:`described below<install-cmdstan-fun>`.

If you installed CmdStanPy with conda, CmdStan and the C++ toolchain,
both CmdStan and the C++ toolchain are installed into directory ``$CONDA_PREFIX/bin``
and you don't need to do any further installs.

C++ Toolchain Requirements
^^^^^^^^^^^^^^^^^^^^^^^^^^

To compile a Stan program requires a modern C++ compiler and the GNU-Make build utility.
These vary by operating system.

**Linux** The required C++ compiler is ``g++ 4.9 3``.
On most systems the GNU-Make utility is pre-installed and is the default ``make`` utility.
There is usually a pre-installed C++ compiler as well, but not necessarily new enough.

**MacOS** The Xcode and Xcode command line tools must be installed.  Xcode is available for free from the Mac App Store.
To install the Xcode command line tools, run the shell command: ``xcode-select --install``.

**Windows**  We recommend using the `RTools 4.0 <https://cran.r-project.org/bin/windows/Rtools/rtools40.html>`_ toolchain

> which contains a ``g++ 8`` compiler and ``Mingw``, the native Windows equivalent of the GNU-Make utility.

Reviewer: ```suggestion
which contains a ``g++ 8`` compiler and ``Mingw-make``, the native Windows equivalent of the GNU-Make utility.
```

> If it is not found on your system, the :meth:`cmdstanpy.install_cmdstan` function will install it automatically.

Reviewer: This is not exactly true - you need to pass `compiler=True`, which isn't quite automatic

Author: looks like command line version doesn't have `compiler=True` arg.  it should.

Author: @WardBrian could you do this in #579?

Reviewer: It does have —compiler, but it only shows up on Windows

Author: great, thanks!

.. code-block:: bash

>     install_cmdstan --compiler True

Reviewer: ```suggestion
    install_cmdstan --compiler
```

The flag doesn't take any arguments, it is similar to `--progress` where passing it is equivalent to `compiler=True` and not passing (default) is False

Author: fixed
